internal/db/db.go

```diff
@@ -50,6 +51,30 @@ func (db *dB) InsertCompose(jobId, accountId, orgId string, request json.RawMess
 	return err
 }
 
+// Returns (true, nil) if the composeId belongs to the accountId (false, nil)
+// otherwise.
+// Returns (false, err) when an error occurs
+func (db *dB) DoesComposesBelongsToAccount(accountId string, composeId string) (
+	bool, error) {
+	ctx := context.Background()
+	conn, err := db.Pool.Acquire(ctx)
+	if err != nil {
+		return false, err
+	}
+	defer conn.Release()
+
+	var count int
+	err = conn.QueryRow(ctx,
+		"SELECT COUNT(*) FROM composes WHERE account_id=$1 AND job_id=$2",
+		accountId,
+		composeId).Scan(&count)
+	if err != nil {
+		return false, err
+	} else {
+		return count > 0, nil
+	}
```

Comment:
With `GetCompose` you can just scan the entire row into a ComposeEntry. And `.Scan` will return `pgx.ErrNoRows` which can easily be checked like `err != nil && errors.As(err, &pgx.ErrNoRows)` in case no row was found. In that case `GetCompose` would return `nil, NoRowsError`.

Reply:
Oh yes ! I'll do that too. Thanks.